### PR TITLE
Include Codecov in Github Actions #51

### LINKED ISSUE #51: Include Codecov in github actions
Include Codecov actions in project workflow
More details here: https://github.com/codecov/codecov-action

### PR DESCRIPTION
**Pull request by tullur**
Create codecov.yml in workflows.
Generate coverage report and upload coverage report with github actions.

Closes #51

### CODE REVIEW

**Comment by rodrigo-brito** [comment]
Hi @tullur,

Thank you for you contribution. We have a working workflow in https://github.com/rodrigo-brito/gocity/blob/master/.github/workflows/tests.yml
Can you include the your setup in this file? To avoid code duplication.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -25,3 +25,12 @@ jobs:
                   go test -race -cover ./...
               env:
                   GOPATH: /home/runner/work/gocity/go
+            - name: Generate coverage report
+              run: |
+                    go test `go list ./...` -race -coverprofile=coverage.txt -covermode=atomic
```
Comment (rodrigo-brito) [inline .github/workflows/tests.yml:30]: The go test command is duplicated. Can you only include the parameters `-coverprofile=coverage.txt -covermode=atomic.` in the previous step?

**Review by rodrigo-brito — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (tullur) [reply .github/workflows/tests.yml:30]: Yep, on my way.

Comment (tullur) [reply .github/workflows/tests.yml]: Done.

**Comment by rodrigo-brito** [comment]
Thank you!

**Comment by tullur** [comment]
Thank you for reviewing @rodrigo-brito

**Pull request merged**